pkgs/jnigen/doc/lifecycle.md

```diff
@@ -0,0 +1,150 @@
+## Lifecycle of Java objects
+
+All Java classes generated by `package:jnigen` extend `JObject`. In JNI there
+are several kinds of references: local, global, and weak global. Local
+references are tied to a single thread. To enable seamless transfer of
+`JObject`s between isolates and safe usage in asynchronous code, `JObject`s
+always use **global** references. It's crucial to remember that there's a limit
+of approximately 50,000 global references.
+
+Once all references (in both Java and Dart) to an object are gone, Java's
+garbage collector (GC) can reclaim it.
+
+### Automatic reference management
+
+`JObject`s attach a native finalizer to their global references. Therefore, when
+the Dart GC collects them, the underlying Java reference is released.
+
+This approach works well for application code where a large number of references
+aren't created repeatedly. However, it's not ideal for packages because the
+usage patterns are unpredictable. This can lead to exceeding the global
+reference limit and application crashes.
+
+### Eagerly releasing references manually (recommended for packages)
+
+Instead of waiting for Dart GC to release the JNI global references,
+`.release()` can be called on the `JObject`s.
+
+```dart
+// Construct the object.
+final hello = 'Hello'.toJString();
+// Use it.
+print(hello);
+// Eagerly release it!
+hello.release();
+```
+
+To make releasing easier, we can use
+[`Arena`](https://pub.dev/documentation/ffi/latest/ffi/Arena-class.html)s.
+First, create an `Arena` via
+[`using`](https://pub.dev/documentation/ffi/latest/ffi/using.html). Then
+register the object to be released at the end of the callback.
+
+```dart
+using((arena) {
+  final hello = 'Hello'.toJString()..releasedBy(arena);
+  final world = 'World'.toJString()..releasedBy(arena);
+  print(hello);
+  print(world);
+});
+// Both `hello` and `world` are now released.
+```
+
+### Lifecycle of interfaces implemented in Dart
+
+Java interfaces can be implemented in Dart using `.implement`. The Dart closure
+implementations must remain accessible as long as the created Java object
+exists. Ownership of these Dart objects is transferred to Java. When the Java GC
+reclaims the implemented object, a message is sent to Dart to remove the
+corresponding closures.
```

Comment:
> The Dart closure implementations must remain accessible as long as the created Java object exists.

This implies to me that me as a user of JNI/JNIGEN should take care of this and make sure it's not GCed.

With the rest of the comment, I'm not so sure though. Could you clarify or maybe rephrase?